### PR TITLE
MINOR: Various code cleanups in core and clients

### PR DESCRIPTION
**Pull request by mimaison**
- Removed unused imports
- Use string interpolations instead of concatenations

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/AdminUtils.scala [hunk 1]
```diff
@@ -473,10 +473,10 @@ object AdminUtils extends Logging with AdminUtilities {
       val jsonPartitionData = zkUtils.replicaAssignmentZkData(replicaAssignment.map(e => e._1.toString -> e._2))
 
       if (!update) {
-        info("Topic creation " + jsonPartitionData.toString)
+        info(s"Topic creation ${jsonPartitionData.toString}")
         zkUtils.createPersistentPath(zkPath, jsonPartitionData)
       } else {
-        info("Topic update " + jsonPartitionData.toString)
+        info(s"Topic update ${jsonPartitionData.toString}")
```
Comment (ijuma) [inline core/src/main/scala/kafka/admin/AdminUtils.scala]: `toString` can be removed in both cases.

File: core/src/main/scala/kafka/server/DelegationTokenManager.scala [hunk 2]
```diff
@@ -317,7 +317,7 @@ class DelegationTokenManager(val config: KafkaConfig,
               tokenInfo.setExpiryTimestamp(expiryTimeStamp)
 
               updateToken(token)
-              info(s"Delegation token renewed for token : " + tokenInfo.tokenId + " for owner :" + tokenInfo.owner)
+              info(s"Delegation token renewed for token : ${tokenInfo.tokenId} for owner :${tokenInfo.owner}")
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/DelegationTokenManager.scala]: Existing issue, but we can maybe fix it now, there should be a space after the `:` not before. A few more cases in this file.

File: core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala [hunk 3]
```diff
@@ -300,7 +300,7 @@ private class ReplicaBuffer(expectedReplicasPerTopicPartition: Map[TopicPartitio
     debug("Begin verification")
     maxLag = -1L
     for ((topicPartition, fetchResponsePerReplica) <- recordsCache) {
-      debug("Verifying " + topicPartition)
+      debug(s"Verifying ${topicPartition}")
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/ReplicaVerificationTool.scala]: No need for braces.

File: clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java [hunk 4]
```diff
@@ -41,7 +41,7 @@
         try (InputStream resourceStream = AppInfoParser.class.getResourceAsStream("/kafka/kafka-version.properties")) {
             props.load(resourceStream);
         } catch (Exception e) {
-            log.warn("Error while loading kafka-version.properties :" + e.getMessage());
+            log.warn("Error while loading kafka-version.properties :{}", e.getMessage());
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java]: Existing issue, space should be after the colon.

File: clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java [hunk 5]
```diff
@@ -106,8 +106,8 @@ private static void unregisterMetrics(Metrics metrics) {
     public static class AppInfo implements AppInfoMBean {
 
         public AppInfo() {
-            log.info("Kafka version : " + AppInfoParser.getVersion());
-            log.info("Kafka commitId : " + AppInfoParser.getCommitId());
+            log.info("Kafka version : {}", AppInfoParser.getVersion());
+            log.info("Kafka commitId : {}", AppInfoParser.getCommitId());
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java]: Existing issue: space after colon. No need for `()` in `getVersion` and `getCommitId`.

**Review by ijuma — COMMENTED** [review]
Thanks for the PRs, just a few nits.

**Comment by mimaison** [comment]
Thanks for the quick review. I've pushed an update

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request merged**